src/lib/field/_selector.scss

```diff
@@ -545,7 +545,7 @@
         @include base.icon-color(invalid);
       }
     }
-    &--disabled {
+    &--disabled:not(.forge-field__addon-end-container--enabled) {
```

Comment:
Is there desire to not have the disabled background color behind this slot?

I ask because it's not immediately obvious that the element is interactive as can be seen here:
![image](https://user-images.githubusercontent.com/2653457/200050289-87cc83e9-a8a2-4c7c-9f41-b14e454c0483.png)

The current DOM structure and styles wouldn't support that too well as I'm sure you noticed, but I'm curious if that was the end goal of the design and the complexity of that change was too high or if this is all we needed to support your usage?